python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py

```diff
@@ -205,3 +210,65 @@ def _mock_interaction():
     else:
         with pytest.raises(Failure, match="failed!"):
             _mock_interaction()
+
+
+def test_fivetran_materialization(
+    connector_id: str,
+    fetch_workspace_data_api_mocks: responses.RequestsMock,
+    sync_and_poll: MagicMock,
+) -> None:
+    with environ({"FIVETRAN_API_KEY": TEST_API_KEY, "FIVETRAN_API_SECRET": TEST_API_SECRET}):
+        workspace = FivetranWorkspace(
+            account_id=TEST_ACCOUNT_ID,
+            api_key=EnvVar("FIVETRAN_API_KEY"),
+            api_secret=EnvVar("FIVETRAN_API_SECRET"),
+        )
+
+        @fivetran_assets(connector_id=connector_id, workspace=workspace, name=connector_id)
+        def my_fivetran_assets(context: AssetExecutionContext, fivetran: FivetranWorkspace):
+            yield from fivetran.sync_and_poll(context=context)
+
+        # Mocked FivetranClient.sync_and_poll returns API response where all connector tables are expected
+        result = materialize(
+            [my_fivetran_assets],
+            resources={"fivetran": workspace},
+        )
+        assert result.success
+        asset_materializations = [
+            event
+            for event in result.events_for_node(connector_id)
+            if event.event_type_value == "ASSET_MATERIALIZATION"
+        ]
+        assert len(asset_materializations) == 4
+        materialized_asset_keys = {
+            asset_materialization.asset_key for asset_materialization in asset_materializations
+        }
+        assert len(materialized_asset_keys) == 4
+        assert my_fivetran_assets.keys == materialized_asset_keys
+
+        # Mocked FivetranClient.sync_and_poll returns API response
```

Comment:
would be nice to have some way of determining that the expected log messages actually happened - this just kinda makes sure it doesn't error. Could potentially directly invoke the fivetran asset and assert against the materializeresult objects + AssetMaterializations when appropriate

Reply:
Done with [capsys](https://docs.pytest.org/en/stable/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function) in [36e9da9](https://github.com/dagster-io/dagster/pull/25961/commits/36e9da9760e7600d24bea6486243cf640f6748a9)